### PR TITLE
feat: openapi auth2 moved to the core service and pipeline decouples openapi

### PR DESCRIPTION
**Pull request by kakj-go**
#### What type of this PR

/kind feature

#### What this PR does / why we need it:
In order to realize the pipeline as the underlying service to decouple the upper-layer services, the auth2 interface is moved to core-services, so that the pipeline calls core-service instead of openapi

#### Which issue(s) this PR fixes:

- [Erda Cloud Issue Link](https://terminus-org.app.terminus.io/erda/dop/projects/387/issues/gantt?filter__urlQuery=eyJpdGVyYXRpb24iOls3NzJdLCJtZW1iZXIiOlsiMTAwMDU2MCJdfQ%3D%3D&id=267051&iterationID=772&pId=267098&type=TASK)

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |      openapi auth2 moved to the core service and pipeline decouples openapi        |
| 🇨🇳 中文    |     openapi auth2 移至 core-services，pipeline 解耦 openapi         |

### CODE REVIEW

**Comment by kakj-go** [comment]
test image:

CICD:

image

image

File: bundle/token.go [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/erda-project/erda/apistructs"
+	"github.com/erda-project/erda/bundle/apierrors"
+	"github.com/erda-project/erda/pkg/http/httputil"
+)
+
+func (b *Bundle) GetCoreServiceOAuth2Token(req apistructs.OpenapiOAuth2TokenGetRequest) (*apistructs.OpenapiOAuth2Token, error) {
```
Comment (sfwn) [inline bundle/token.go:28]: reuse original bundle and just replace openapi to core-services.

Comment (sfwn) [reply bundle/token.go:28]: and rename to GetOAuth2Token, remove module prefix

**Review by sfwn — CHANGES_REQUESTED** [review]

File: bundle/openapi.go [hunk 2]
```diff
@@ -15,68 +15,14 @@
...
-	"errors"
-	"fmt"
-
 	"github.com/erda-project/erda/apistructs"
-	"github.com/erda-project/erda/bundle/apierrors"
-	"github.com/erda-project/erda/pkg/http/httputil"
 )
 
 // applyOpenAPIToken 从 openapi 动态获取 oauth2 token
 func (b *Bundle) GetOpenapiOAuth2Token(req apistructs.OpenapiOAuth2TokenGetRequest) (*apistructs.OpenapiOAuth2Token, error) {
```
Comment (sfwn) [inline bundle/openapi.go:22]: no need this method anymore.
just rename to GetOAuth2Token, and adjust other related structs pls.

File: bundle/token.go [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/erda-project/erda/apistructs"
+	"github.com/erda-project/erda/bundle/apierrors"
+	"github.com/erda-project/erda/pkg/http/httputil"
+)
+
+func (b *Bundle) GetOAuth2Token(req apistructs.OpenapiOAuth2TokenGetRequest) (*apistructs.OpenapiOAuth2Token, error) {
```
Comment (sfwn) [inline bundle/token.go:28]: delete these two funcs.

**Review by sfwn — CHANGES_REQUESTED** [review]

**Review by sfwn — APPROVED** [review]

**Review by sfwn — APPROVED** [review]

**Pull request merged**